### DISCUSSION TITLE
Is there a way to dynamically add new websites?

### DISCUSSION THREAD

**Discussion by ozziestraw** [umami-software/umami#625] [Q&A]
Can we dynamically add a new website on the go, i.e. through an API?

Can this support unlimited websites?

Thanks!

**Comment by gnarlex** [comment]
I'm gonna quote @mikecao for this one (https://github.com/mikecao/umami/issues/372#issuecomment-727557082):
> The app itself uses API calls to get data. It's not documented but you can watch your network requests to make the same calls or look through the source code.

So yeah you can dynamically create websites. This can be done via a HTTP `POST` request to `/api/website`. I suggest you study [the source code for that endpoint](https://github.com/mikecao/umami/blob/master/pages/api/website/index.js), if you're interested in that.

On that note: Adding API documentation would be useful for developers.
